### PR TITLE
aws-c-*: bump to latest

### PR DESCRIPTION
**Pull request by r-burns**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
@ofborg eval

File: pkgs/development/libraries/s2n-tls/default.nix [hunk 1]
```diff
@@ -7,23 +7,15 @@
...
     owner = "aws";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-6XqBpNURU8fzGkTt4jsijgMiOkzMebmLmPAq8yQsTg4=";
+    sha256 = "sha256-gd91thIcJO6Bhn1ENkW0k2iDzu1CvSYwWVv0VEM9umU=";
   };
 
-  patches = [
-    # Fix FindLibCrypto paths (https://github.com/aws/s2n-tls/pull/3067)
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/s2n-tls/default.nix:21]: Please remove the unused fetchpatch from inputs.

Comment (r-burns) [reply pkgs/development/libraries/s2n-tls/default.nix:21]: Done!

**Comment by r-burns** [comment]
Result of `nixpkgs-review pr 146824` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>ovftool</li>
  </ul>
</details>
<details>
  <summary>3 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
    <li>tests.nixos-functions.nixosTest-test</li>
  </ul>
</details>
<details>
  <summary>107 packages built:</summary>
  <ul>
    <li>arrow-cpp</li>
    <li>aws-c-auth</li>
    <li>aws-c-cal</li>
    <li>aws-c-common</li>
    <li>aws-c-compression</li>
    <li>aws-c-event-stream</li>
    <li>aws-c-http</li>
    <li>aws-c-io</li>
    <li>aws-c-mqtt</li>
    <li>aws-c-s3</li>
    <li>aws-c-sdkutils</li>
    <li>aws-checksums</li>
    <li>aws-crt-cpp</li>
    <li>aws-sdk-cpp</li>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>cached-nix-shell</li>
    <li>cachix</li>
    <li>common-updater-scripts</li>
    <li>crate2nix</li>
    <li>crystal2nix</li>
    <li>dep2nix</li>
    <li>dydisnix</li>
    <li>fusionInventory</li>
    <li>go2nix</li>
    <li>haskellPackages.cabal2nix-unstable</li>
    <li>haskellPackages.cachix</li>
    <li>haskellPackages.hercules-ci-agent</li>
    <li>haskellPackages.hercules-ci-cli</li>
    <li>haskellPackages.hercules-ci-cnix-expr</li>
    <li>haskellPackages.hercules-ci-cnix-store</li>
    <li>haskellPackages.nix-paths</li>
    <li>haskellPackages.nvfetcher</li>
    <li>haskellPackages.update-nix-fetchgit</li>
    <li>hci</li>
    <li>hercules-ci-agent</li>
    <li>hubstaff</li>
    <li>hydra-unstable</li>
    <li>libnixxml</li>
    <li>lispPackages.quicklisp-to-nix</li>
    <li>lispPackages.quicklisp-to-nix-system-info</li>
    <li>nix (nixStable ,nix_2_4)</li>
    <li>nix-bundle</li>
    <li>nix-direnv</li>
    <li>nix-doc</li>
    <li>nix-du</li>
    <li>nix-eval-jobs</li>
    <li>nix-index</li>
    <li>nix-pin</li>
    <li>nix-plugins</li>
    <li>nix-prefetch</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-cvs</li>
    <li>nix-prefetch-docker</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-hg</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-serve</li>
    <li>nix-simple-deploy</li>
    <li>nix-template</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nixUnstable</li>
    <li>nix_2_3</li>
    <li>nixos-generators</li>
    <li>nixos-option</li>
    <li>nixos-rebuild</li>
    <li>nixos-shell</li>
    <li>nixpkgs-review</li>
    <li>nixui</li>
    <li>nvfetcher</li>
    <li>prefetch-yarn-deps</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.intake-parquet</li>
    <li>python38Packages.nix-kernel</li>
    <li>python38Packages.nixpkgs</li>
    <li>python38Packages.pyarrow</li>
    <li>python38Packages.pythonix</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.intake-parquet</li>
    <li>python39Packages.nix-kernel</li>
    <li>python39Packages.nixpkgs</li>
    <li>python39Packages.pyarrow</li>
    <li>python39Packages.pythonix</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
    <li>rnix-lsp</li>
    <li>s2n-tls</li>
    <li>terranix</li>
    <li>tests.trivial-builders.references</li>
    <li>update-nix-fetchgit</li>
    <li>vgo2nix</li>
    <li>vulnix</li>
    <li>yarn2nix</li>
  </ul>
</details>

**Pull request merged**